src/SDKs/SqlManagement/Sql.Tests/Utilities/TestEnvironmentUtilities.cs

```diff
@@ -9,6 +9,75 @@ namespace Sql.Tests
 {
     public static class TestEnvironmentUtilities
     {
+        private const string _environmentVariableName = "TEST_CSM_ORGID_AUTHENTICATION";
+
+        private static readonly TestEnvironment _environment = 
+            new TestEnvironment(Environment.GetEnvironmentVariable(_environmentVariableName));
+
+        // We now load default locations from environment variable.
+        // Tests were recorded with this appended to DefaultLocationId=japaneast;DefaultLocation=Japan East;DefaultSecondaryLocationId=centralus;DefaultSecondaryLocation=Central US;DefaultStagePrimaryLocation=North Europe;DefaultStageSecondaryLocation=SouthEast Asia;DefaultEuapPrimaryLocation=East US 2 EUAP;DefaultEuapPrimaryLocationId=eastus2euap
+        public static string DefaultLocation
+        {
+            get
+            {
+                return GetValueFromEnvironment("DefaultLocation");
```

Comment:
It'll be very easy for our users to find out that these variables are required, as they will get an error that tells them exactly what key is missing and what variable it should be in, so that's not a concern.

You had told me in a previous email to hard-code the location in the connection string.  Why did this change?  This random location approach has never been mentioned and we haven't agreed to it.

"From: Abhijeet Shah 
Sent: Monday, July 17, 2017 12:49 PM
To: Nathan Fan <nafan@microsoft.com>; Jared Moore <Jared.Moore@microsoft.com>; Nayantara Duttachoudhury <naduttac@microsoft.com>
Subject: RE: PR pending for fixing virtual network rules tests

Hello Nathan,
Ok my bad, I gave you the link to the wrong TestFx feature.
That link is not correct, so ignore that.

So what I meant was, with the existing connection string that you pass in to the test using TEST_CSM_ORGID_AUTHENTICATION
You can now pass in custom key-value paris
e.g. TEST_CSM_ORGID_AUTHENTICATION=SubscriptionId={SubId};UserId={orgId};AADTenant={tenantId};Environment={env};MyLocation=MyHardCodedLocation
If you provide the above mentioned connection string, you can now inject the hardcodedLocation string using
String location = TestEnvironment.ConnectionString.KeyValuePairs["MyLocation"]

So even though we allow you to inject custom data during tests, the person who records the same test (with “MyLocation”) will need to know that this was recorded using a custom region, hence we request you to make all your test Region agnostic.

Abhijeet"